pkg/scaffold/cmd.go

```diff
@@ -121,6 +124,15 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Expose metrics by creating a Service object
+	s, err := metrics.ExposeMetricsPort(metricsAddress)
```

Comment:
Now might be a good time to consider if we need both `ExposeMetricsPort(metricsAddress)` and `InitOperatorService(port, portName)` as public functions since the former is just a wrapper around the latter. Or if we should rename the two into one.

Also as I pointed out above, we probably want to skip creating the service if we're running locally.
I know we previously decided to just have `up local` create the service https://github.com/operator-framework/operator-sdk/issues/462#issuecomment-428293427 
However since we've decided to GC the metrics service(by setting ownership to the pod) #459  we'll want to skip creating it altogether when running locally.

In which case where do we skip creating the service once we know the operator is running locally via the operator namespace [`myNS()`](https://github.com/operator-framework/operator-sdk/blob/master/pkg/leader/leader.go#L148) check. We could return the error here and skip the call to `client.Create()`.
Or move the `client.Create()` call back into the function and skip it there.

Reply:
> Now might be a good time to consider if we need both ExposeMetricsPort(metricsAddress) and InitOperatorService(port, portName) as public functions since the former is just a wrapper around the latter. Or if we should rename the two into one.

For the operator specific metrics (k-s-m) we will need to create a Service object and expose the port, so we will need to use the function there. But I am fine with moving it into metrics pkg as it's not used anywhere else and it is exposing metrics always. Up to you?

Reply:
We can move `InitOperatorService()` to the metrics pkg as a private helper. Unless we need to provide this as a separate helper to the user we can keep it internal to the SDK.